connectivity/FEATURE_BLE/source/common/ble_trace_helpers.h

```diff
@@ -37,6 +44,11 @@ static inline const char* tr_as_array(T item)
     return (mbed_trace_array)((const uint8_t*)&item, sizeof(item));
 }
 
+static inline const char* to_string(UUID uuid)
+{
+    return (mbed_trace_array)(uuid.getBaseUUID(), uuid.getLen());
+}
+
```

Comment:
This function does not modify `uuid`, i.e. you're only using getters. The parameter could be a constant reference: `const UUID &uuid`. Also, using `mbed_trace_array` will cause the UUID to be printed in reverse because the bytes are stored in little endian in the array returned by `getBaseUUID`. There's also the (_non_-)issue of the colon delimiters between each byte in the trace. Personally, I think this should be the way to print UUIDs but we need to add some flexibility to [mbed_trace](https://github.com/ARMmbed/mbed-os/tree/master/platform/mbed-trace).

What do you think @pan- ?

Reply:
you can replace this when you write your to_string function for UUID we talked about